packages/@aws-cdk/aws-iam/lib/lazy-policy.ts

```diff
@@ -0,0 +1,112 @@
+import { Aws, CfnRefElement, Construct, Lazy, Resource } from '@aws-cdk/core';
+import { IGroup } from './group';
+import { CfnPolicy } from './iam.generated';
+import { IPolicy, Policy, PolicyProps } from './policy';
+import { PolicyStatement } from './policy-statement';
+import { IRole } from './role';
+import { IUser } from './user';
+
+export class LazyPolicy extends Resource implements IPolicy {
```

Comment:
Might need a docstring explaining what it does.

Does it work?

Reply:
It does not currently, because of the magical overrides of `ref` and `logicalId` of the custom L1 here: https://github.com/aws/aws-cdk/pull/4501/files?file-filters%5B%5D=.ts#diff-f60a11b2b65df07bd241981fe0a626d7R37-R55